packages/devtools/lib/src/memory/memory.dart

```diff
@@ -362,38 +650,291 @@ class MemoryScreen extends Screen with SetStateMixin {
-      try {
-        instance = await memoryController.getObject(row.objectRef);
-      } catch (e) {
-        instance = null; // Signal a problem
-      } finally {
-        tableContainer.add(_createInstanceView(
-          instance != null
-              ? row.objectRef
-              : 'Unable to fetch instance ${row.objectRef}',
-          row.className,
-        ));
-
-        tableContainer.element.scrollTo(<String, dynamic>{
-          'left': tableContainer.element.scrollWidth,
-          'top': 0,
-          'behavior': 'smooth',
-        });
+  void select(InstanceSummary row) async {
+    ga.select(ga.memory, ga.inspectInstance);
+
+    // User selected a new instance from the list of class instances so the
+    // instance view which would be the third child needs to be removed.
+    _removeInstanceView();
 
-        // Allow inspection of the memory object.
-        memoryDataView.showFields(instance != null ? instance.fields : []);
+    Instance instance;
+    try {
+      instance = await memoryController.getObject(row.objectRef);
+
+      final dynamic theObject = await memoryController.getObject(row.objectRef);
+      if (theObject is Instance) {
+        instance = theObject;
+      } else if (theObject is Sentinel) {
+        instance = null;
+        // TODO(terry): Tracking Sentinel's to be removed.
+        framework.toast('Sentinel ${row.objectRef}', title: 'Warning');
       }
-    });
+    } catch (e) {
+      instance = null; // Signal a problem
+    } finally {
+      tableContainer.add(_createInstanceView(
+        instance != null
+            ? row.objectRef
+            : 'Unable to fetch instance ${row.objectRef}',
+        row.className,
+      ));
 
-    return table;
+      tableContainer.element.scrollTo(<String, dynamic>{
+        'left': tableContainer.element.scrollWidth,
+        'top': 0,
+        'behavior': 'smooth',
+      });
+
+      // Allow inspection of the memory object.
+      memoryDataView.showFields(instance != null ? instance.fields : []);
+
+      // Record this navigation.
+      final InstanceRef hashCodeResult =
+          await evaluate(row.objectRef, 'hashCode');
+
+      if (!fromMemoryHover &&
+          (memoryPath.isLastInBound || memoryPath.isLastInstance)) {
+        // User clicked an instance, start new history with this instance.
+        _resetHistory();
+      }
+
+      // Record the memory navigation.
+      memoryPath.add(NavigationState.instanceSelect(
+        row.className,
+        hashCodeResult.valueAsString,
+      ));
+
+      if (memoryPath.isLastInBound) {
+        // Re-construct memory navigation and display.
+        history.clear();
+        memoryPath.addPathsLinks(history, _handleHistoryClicks);
+
+        history..clazz('history-navigation', removeOthers: true);
+      }
+    }
+  }
+
+  void _handleHistoryClicks(CoreElement element) {
+    // Handle clicking in the history links.
+    if (element.tag == 'SPAN' && element.hasClass('history-link')) {
```

Comment:
nit: just check the class and don't worry about the tag.

Reply:
Done, added assert for tag check.